src/System.Linq/src/System/Linq/Repeat.cs

```diff
@@ -99,6 +99,18 @@ public List<TResult> ToList()
                 return list;
             }
 
+            public HashSet<TResult> ToHashSet(IEqualityComparer<TResult> comparer)
+            {
+                HashSet<TResult> hashSet = new HashSet<TResult>(_count, comparer);
+
+                for (int i = 0; i != _count; ++i)
+                {
+                    hashSet.Add(_current);
+                }
```

Comment:
@TiagoSoczek A comparer that says the exact same object isn't equal to itself is fundamentally broken. It's better to keep the implementation simple and not penalize the non-broken use case.